File: FestivalC#/Festival/Model/Model/Transaction.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Festival.Model
{
    public class Transaction
    {
        //id
        private int idTransaction;

        //name of the client
        private String clientName;

        //number of tickets sold 
        private int numberOfTickets;

        //id of the show
        private Show show;

        //Constructor
        public Transaction(int idTransaction, String clientName, int numberOfTickets, Show show)
        {
            this.idTransaction = idTransaction;
            this.clientName = clientName;
            this.numberOfTickets = numberOfTickets;
            this.Show = show;
        }

        //Constructor with default id 
        public Transaction(String clientName, int numberOfTickets, Show show)
        {
            this.idTransaction = 1;
            this.clientName = clientName;
            this.numberOfTickets = numberOfTickets;
            this.Show = show;
        }
        

        public int IdTransaction
        {
            get
            {
                return idTransaction;
            }

            set
            {
                idTransaction = value;
            }
        }

        public string ClientName
        {
            get
            {
                return clientName;
            }

            set
            {
                clientName = value;
            }
        }

        public int NumberOfTickets
        {
            get
            {
                return numberOfTickets;
            }

            set
            {
                numberOfTickets = value;
            }
        }

        public Show Show
        {
            get
            {
                return show;
            }

            set
            {
                show = value;
            }
        }

        public override bool Equals(object obj)
        {
            if (obj == null)
            {
                return false;
            }
            if (!(obj is Transaction))
            {
                return false;
            }
            Transaction t = (Transaction)obj;
            return t.idTransaction == idTransaction && t.clientName.Equals(clientName)
                && t.numberOfTickets == numberOfTickets && t.show.Equals(show);
        }

        public override string ToString()
        {
            return idTransaction + " " + clientName + " " + numberOfTickets + " " + show.IdShow;
        }

        public override int GetHashCode()
        {
            return base.GetHashCode();
        }
    }
}

File: FestivalJava/Model/src/main/java/Domain/Transaction.java
package Domain;

import java.io.Serializable;

/**
 * Created by Sebi on 09-Mar-17.
 */
public class Transaction implements Serializable {
    //id
    private Integer idTransaction;

    //name of client
    private String clientName;

    //number of tickets bought
    private Integer numberOfTickets;

    //show for which the tickets were bought
    private Show show;

    //Constructor
    public Transaction(Integer idTransaction, String clientName, Integer numberOfTickets, Show show){
        this.idTransaction = idTransaction;
        this.clientName = clientName;
        this.numberOfTickets = numberOfTickets;
        this.show = show;
    }

    //Constructor with default id
    public Transaction(String clientName, Integer numberOfTickets, Show show){
        this.idTransaction = 0;
        this.clientName = clientName;
        this.numberOfTickets = numberOfTickets;
        this.show = show;
    }


    //getters and setters
    public Integer getIdTransaction() {
        return idTransaction;
    }

    public void setIdTransaction(Integer idTransaction) {
        this.idTransaction = idTransaction;
    }

    public String getClientName() {
        return clientName;
    }

    public void setClientName(String clientName) {
        this.clientName = clientName;
    }

    public Integer getNumberOfTickets() {
        return numberOfTickets;
    }

    public void setNumberOfTickets(Integer numberOfTickets) {
        this.numberOfTickets = numberOfTickets;
    }


    @Override
    public String toString(){
        return idTransaction + " " + clientName + " " +
                numberOfTickets + " " + show.getIdShow();
    }

    public Show getShow() {
        return show;
    }

    public void setShow(Show show) {
        this.show = show;
    }
}
